addons/product/static/src/js/product_pricelist_report.js

```diff
@@ -1,317 +1,241 @@
-odoo.define('product.generate_pricelist', function (require) {
-'use strict';
-
-var AbstractAction = require('web.AbstractAction');
-var core = require('web.core');
-var FieldMany2One = require('web.relational_fields').FieldMany2One;
-var StandaloneFieldManagerMixin = require('web.StandaloneFieldManagerMixin');
-var Widget = require('web.Widget');
-
-var QWeb = core.qweb;
-var _t = core._t;
-
-var QtyTagWidget = Widget.extend({
-    template: 'product.report_pricelist_qty',
-    events: {
-        'click .o_remove_qty': '_onClickRemoveQty',
-    },
-    /**
-     * @override
-     */
-    init: function (parent, defaulQuantities) {
-        this._super.apply(this, arguments);
-        this.quantities = defaulQuantities;
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { sprintf } from "@web/core/utils/strings";
+import { useService } from "@web/core/utils/hooks";
+import { useSetupAction } from "@web/webclient/actions/action_hook";
+import { Layout } from "@web/search/layout";
+
+const { Component, useState, onWillStart, markup, onRendered } = owl;
+
+
+export class ProductPricelistReport extends Component {
+    setup() {
+        this.action = useService("action");
+        this.orm = useService("orm");
+
         this.MAX_QTY = 5;
-    },
-
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
-
-    /**
-     * Add a quantity when add(+) button clicked.
-     *
-     * @private
-     */
-    _onClickAddQty: function () {
+        const pastState = this.props.state || {};
+
+        // Putting everything that needs to be reactive on the useState wrapper.
+
+        // While activeIds and activeModel are not necessarily reactive we need
+        // them to be keeped for the useSetupAction going back and forward on links.
+        this.state = useState({
+            activeIds: pastState.active_ids || this.props.action.context.active_ids,
+            activeModel: pastState.active_model || this.props.action.context.active_model,
+            displayPricelist: pastState.displayPricelist || false,
+            html: markup(""),
```

Comment:
this isn't necessary

Reply:
@aab-odoo not necessary in the first declaration or everywhere ? because I tried without markup and the rendering of the page was not good

Reply:
The content here isn't html so it's useless to markup it